assayist/processor/utils.py

```diff
@@ -0,0 +1,83 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import os
+import subprocess
+import shutil
+
+import koji
+
+from assayist.processor.configuration import config
+from assayist.processor.logging import log
+
+
+def get_brew_session():
+    """
+    Generate a Brew session.
+
+    :return: a Brew session
+    :rtype: koji.ClientSession
+    """
+    return koji.ClientSession(config.brew_url)
+
+
+def _assert_command(cmd_name):
+    """
+    Ensure a command is installed and can be found using the paths in $PATH.
+
+    :raises RuntimeError: if the command is not installed
+    """
+    if not shutil.which(cmd_name):
+        raise RuntimeError(f'The command "{cmd_name}" is not installed and is required')
+
+
+def download_build(build_identifier, output_dir):
+    """
+    Download the artifacts associated with a Brew build.
+
+    :param str/int build_identifer: the string of the builds NVR or the integer of the build ID
+    :param str output_dir: the path to download the the archives to
+    :return: a list of paths to the downloaded archives
+    :rtype: list
+    """
+    # Make sure the Brew command is installed
+    _assert_command('brew')
+    if not os.path.isdir(output_dir):
+        raise RuntimeError(f'The passed in directory of "{output_dir}" does not exist')
+
+    session = get_brew_session()
+    build = session.getBuild(build_identifier)
+    if not build:
+        raise RuntimeError(f'The Brew build "{build_identifier}" does not exist')
+
+    # There's no API for this, so it's better to just call the CLI directly
+    cmd = ['brew', 'download-build', str(build['id'])]
```

Comment:
The command to use should be configurable, or else it should be `'koji', '--profile', ...`

Reply:
Good point. I changed the PR to use profiles.